rust_src/src/marker.rs

```diff
@@ -70,3 +80,23 @@ pub fn marker_buffer(marker: LispObject) -> LispObject {
         None => LispObject::constant_nil(),
     }
 }
+
+/// Set PT from MARKER's clipped position.
+pub fn set_point_from_marker(marker: LispObject) {
+    let cur_buf = ThreadState::current_buffer();
+    let m = marker.as_marker_or_error();
+    let charpos = clip_to_bounds(
+        cur_buf.begv,
+        marker_position(marker).as_fixnum().unwrap(),
```

Comment:
`maker_position` just wraps charpos in a lisp_object that you unwrap again here :)

Reply:
I tried it without, but it doesn't work.